scripts/addons.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Elico Corp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import os
+import re
+from urlparse import urlparse
+from subprocess import call
+from subprocess import check_output
+
+EXTRA_ADDONS_PATH = '/mnt/data/additional_addons/'
+ODOO_CONF = '/opt/odoo/etc/odoo.conf'
+ADDONS_PATH = ['/opt/odoo/sources/odoo/addons']
+SCHEME = 'https://'
+GIT_REPOSITORY_HOSTING_SERVICE = 'github.com'
+DEFAULT_ORGANIZATION = 'OCA'
+
+
+class Repo(object):
+    """
+    oca_dependencies.txt
+
+    For public repo:
+
+    oca-repo -> https://github.com/OCA/oca-repo (branch: master)
+    oca-repo 8.0 -> https://github.com/OCA/oca-repo (branch: 8.0)
+    organization/public-repo -> https://github.com/organization/public-repo (branch: master)
+    organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    https://github.com/organization/public-repo -> https://github.com/organization/public-repo (branch: master)
+    https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    public-repo_rename https://github.com/organization/public-repo -> https://github.com/organization/public-repo (branch: master)
+    public-repo_rename https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+
+    For private repo:
+
+    git@github.com:Elico-Corp/private_repo
+    git@github.com:Elico-Corp/private_repo 8.0
+    private_repo_rename git@github.com:Elico-Corp/private-repo
+    private_repo_rename git@github.com:Elico-Corp/private_repo 8.0
+
+    """
+    def __init__(self, dependent, parent=None):
+        if parent:
+            self.parent = parent
+            self.branch = self.parent.branch
```

Comment:
Just to make sure: if the branch is specified, is it going to be overridden by the parent branch (because we don't want that)? Or is it the contrary (defined branch will override parent branch during parsing)?